Using meta checks for unary `torch.all` and `torch.any`.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60364 Port `argmin` kernel to structured kernels.
* #60363 Port `argmax` to structured kernels.
* **#60362 Using meta checks for unary `torch.all` and `torch.any`.**

This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

Differential Revision: [D29265858](https://our.internmc.facebook.com/intern/diff/D29265858)

ysiraichi committed:
Using meta checks for unary `torch.all` and `torch.any`.

This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main.

[ghstack-poisoned]

ysiraichi committed:
Update on "Using meta checks for unary `torch.all` and `torch.any`."


This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29265858).

ysiraichi committed:
Update on "Using meta checks for unary `torch.all` and `torch.any`."


This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

Differential Revision: [D29265858](https://our.internmc.facebook.com/intern/diff/D29265858)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29265858).

ezyang committed:
Update on "Using meta checks for unary `torch.all` and `torch.any`."


This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

Differential Revision: [D29265858](https://our.internmc.facebook.com/intern/diff/D29265858)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29265858).

ysiraichi committed:
Update on "Using meta checks for unary `torch.all` and `torch.any`."


This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

Differential Revision: [D29265858](https://our.internmc.facebook.com/intern/diff/D29265858)

[ghstack-poisoned]

ezyang committed:
Update on "Using meta checks for unary `torch.all` and `torch.any`."


This PR makes use of the newly implemented unified `at::meta::check_reduction` for
validating the inputs and configuring its `TensorIterator`.

This PR was openned so as to solve the CI failures in main when merging: #59371 #59372 #59373 #59937 #59938.

Differential Revision: [D29265858](https://our.internmc.facebook.com/intern/diff/D29265858)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29265858).
